scripts/generate_minknow_samplesheet.py

```diff
@@ -0,0 +1,519 @@
+#!/usr/bin/env python
+
+import logging
+import os
+import re
+import shutil
+import sys
+from argparse import ArgumentParser
+from datetime import datetime as dt
+
+import pandas as pd
+from genologics.config import BASEURI, PASSWORD, USERNAME
+from genologics.entities import Artifact, Process
+from genologics.lims import Lims
+from tabulate import tabulate
+
+from data.ONT_barcodes import ONT_BARCODE_LABEL_PATTERN, ONT_BARCODES
+from epp_utils.udf_tools import fetch
+from scilifelab_epps.epp import traceback_to_step, upload_file
+
+DESC = """ Script to generate MinKNOW samplesheet for starting ONT runs.
+"""
+
+TIMESTAMP = dt.now().strftime("%y%m%d_%H%M%S")
+SCRIPT_NAME: str = os.path.basename(__file__).split(".")[0]
+
+
+def get_ont_library_contents(
+    ont_library: Artifact,
+    list_contents: bool = False,
+    print_dataframe: bool = False,
+) -> pd.DataFrame:
+    """For an ONT sequencing library, compile a dataframe with sample-level information.
+
+    Will backtrack the library to previous pooling step (if any) to elucidate
+    sample and index information and decide whether to demultiplex at the level of
+    ONT barcodes, Illumina indices, both or neither.
+
+    """
+
+    ont_barcode_well2label = {}
+    for ont_barcode_dict in ONT_BARCODES:
+        ont_barcode_well2label[ont_barcode_dict["well"]] = ont_barcode_dict["label"]
+
+    logging.info(
+        f"Compiling sample-level information for library '{ont_library.name}'..."
+    )
+    library_contents_msg = f"ONT sequencing library '{ont_library.name}' consists of:"
+
+    # Instantiate list to collect dataframe rows
+    rows = []
+
+    # Remaining possibilities:
```

Comment:
👍

Reply:
Is this referring to the comments showing the shrinking list of possibilities or something else?